modules/sdk/gradle-plugins-defaults/src/main/java/com/liferay/gradle/plugins/defaults/tasks/SetBuildProfileTask.java

```diff
@@ -0,0 +1,295 @@
+			sb.append("Profile name cannot contain ',' or ':'.");
+			sb.append(System.lineSeparator());
+
+			throw new IllegalArgumentException(sb.toString());
+		}
+
+		Logger logger = getLogger();
+
+		logger.lifecycle("Setting build profile name: {}", profileName);
+
+		Collection<Project> projectDependencies = _getProjectDependencies(
+			project);
+
+		for (Project projectDependency : projectDependencies) {
+			File lfrBuildFile = new File(
+				projectDependency.getProjectDir(), LFRBUILD_PROFILES_FILENAME);
+
+			try {
+				_processBuildFile(profileName, logger, lfrBuildFile);
+			}
+			catch (IOException ioe) {
+				throw new GradleException(
+					"Unable to create build profile custom marker:" +
+						lfrBuildFile.getAbsolutePath(),
+					ioe);
+			}
+		}
+
+		StringBuilder sb = new StringBuilder();
+
+		sb.append("Build profile " + profileName + " created successfully.");
+		sb.append(System.lineSeparator());
+		sb.append("To import or use this profile, ");
+		sb.append("please use the following argument: ");
+		sb.append(System.lineSeparator());
+		sb.append("-D");
+		sb.append(LiferaySettingsPlugin.BUILD_PROFILE_PROP_KEY);
+		sb.append("=" + profileName);
+
+		if (profileName.contains(":")) {
+			String[] path = profileName.split(":");
+
+			String simpleProfileName = path[path.length - 1];
+
+			sb.append("The value can also be specified as: ");
+			sb.append(System.lineSeparator());
+			sb.append("-D");
+			sb.append(LiferaySettingsPlugin.BUILD_PROFILE_PROP_KEY);
+			sb.append("=" + simpleProfileName);
+		}
+
+		sb.append(System.lineSeparator());
+
+		sb.append(
+			"The value may be comma separated to specify multiple profiles.");
+
+		String message = sb.toString();
+
+		logger.lifecycle(message);
+	}
+
+	private static void _processBuildFile(
+			String profileName, Logger logger, File lfrBuildFile)
+		throws FileNotFoundException, IOException {
+
+		ArrayList<String> list = new ArrayList<>();
+
+		boolean missingProfile = false;
+
+		if (lfrBuildFile.exists()) {
+			logger.lifecycle("Reading {}", lfrBuildFile);
+
+			try (Scanner s = new Scanner(lfrBuildFile)) {
+				while (s.hasNext()) {
+					String foundProfileName = s.next();
+
+					if (!foundProfileName.isEmpty()) {
+						if (!list.contains(foundProfileName)) {
+							list.add(foundProfileName);
+						}
+					}
+				}
+			}
+
+			missingProfile = !list.contains(profileName);
+
+			if (missingProfile) {
+				lfrBuildFile.delete();
+			}
+		}
+		else {
+			missingProfile = true;
+		}
+
+		if (missingProfile) {
+			list.add(profileName);
+		}
+
+		if (missingProfile) {
+			boolean buildFileExists = lfrBuildFile.exists();
+
+			try (FileWriter fw = new FileWriter(
+					lfrBuildFile, buildFileExists)) {
+
+				if (buildFileExists) {
+					fw.write(System.lineSeparator());
+				}
+
+				for (String foundProfileName : list) {
+					fw.write(foundProfileName + System.lineSeparator());
+				}
+			}
+		}
+	}
+
+	private Collection<Project> _getProjectDependencies(Project project) {
```

Comment:
This should be unified with the CleanBuildProfileTask _getProjectDependencies, since they are sharing much of the same code. Probably should move to a Util class.